SharpGen.Platform/SharpGen.Platform.csproj

```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
What's the reasoning for splitting out SharpGen.Platform?

Reply:
To drop runtime-specific dependencies from `SharpGen.Runtime`. RID-specific assembly lookup is highly error-prone (sometimes it just doesn't work, for example when MSBuild ALCs are used). This split removes RID-specific assemblies from SharpGen plugins' (e.g. SG.Doc.MSDN) dependency graph.
It also paves the road for the migration off the CastXML by creating interface and implementation parts. The interface is not a complete C++ parser abstraction, rather, it is a first draft of such.